File: Recursion/increasing.cpp
// #include<iostream>
// using namespace std;

// void increasing(int n)
// {
//     if(n == 0)
//     {
//         return;
//     }
//     for(int i = 0; i <= n; i++)
//     {
//         cout << i << endl;
//     }
// }

// int main()
// {
//     int n;
//     cin >> n;
//     increasing(n);
//     return 0;
// }

#include<iostream>
using namespace std;
void increasing(int n)
{
    if(n == 0)
    {
        return;
    }
    increasing(n - 1);
    cout << n << endl;
}

int main()
{
    int n;
    cin >> n;
    increasing(n);
    return 0;
}

File: Recursion/increasing.java
import java.util.Scanner;
public class increasing
{
    public static Scanner scn = new Scanner(System.in);
    public static void increasing(int n)
    {
        if(n == 0)
        {
            return;
        }
        increasing(n - 1);
        System.out.println(n);
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        increasing(n);
    }
}